test/ecdsa_adaptor.json

```diff
@@ -0,0 +1,56 @@
+    },
+    {
+        "adaptor_sig" : "024288a3e3d08ac29bab9a6396a0c4c2a74a923dbaca92c469fe38eb0bfde131b203127e25e61da3a9afbb0708eba169c1454f2a12785e3bd371b1a070d37c856b4a5adc9e4e988b4d30470fc94b838ebc202f6bf5fe66805cb957c89275cee34139d05ad315068f38bce87ffe269fed4cfb68438a536e64bda2d6bac695038b2499348a2a6fed030aa94952e36aaf002657e3ad5b1da9a2b90055675468e0011ee5",
+        "message_hash": "8131e6f4b45754f2c90bd06688ceeabc0c45055460729928b4eecf11026a9e2d",
+        "public_signing_key" : "03ef16dd7c75ca40cfeab2aa659f2201e857591df3de67494a4d1dae34587395e6",
+        "encryption_key" : "0254425b8f73754f16f758937fdab39a04ea61763f9c8b2ba8abe082882ddbf155",
+        "decryption_key" : "7f82dd63f4f75c33da444b72372be3aa43c0027a076bf9675eb7932695d127a4",
+        "signature" : "4288a3e3d08ac29bab9a6396a0c4c2a74a923dbaca92c469fe38eb0bfde131b21321d75e2d155437f4f5b918c9becb327096dd5e72343f39c51565b6110f78be",
+        "comment" : "valid adaptor signature #2"
+    },
+    {
+        "adaptor_sig" : "024288a3e3d08ac29bab9a6396a0c4c2a74a923dbaca92c469fe38eb0bfde131b203127e25e61da3a9afbb0708eba169c1454f2a12785e3bd371b1a070d37c856b4aa52361b16774b2cfb8f036b47c7143de8b42e6e848c843826809cc1701530008d05ad315068f38bce87ffe269fed4cfb68438a536e64bda2d6bac695038b2499348a2a6fed030aa94952e36aaf002657e3ad5b1da9a2b90055675468e0011ee5",
+        "message_hash": "8131e6f4b45754f2c90bd06688ceeabc0c45055460729928b4eecf11026a9e2d",
+        "public_signing_key" : "03ef16dd7c75ca40cfeab2aa659f2201e857591df3de67494a4d1dae34587395e6",
+        "encryption_key" : "0254425b8f73754f16f758937fdab39a04ea61763f9c8b2ba8abe082882ddbf155",
+        "decryption_key" : "7f82dd63f4f75c33da444b72372be3aa43c0027a076bf9675eb7932695d127a4",
+        "signature" : "4288a3e3d08ac29bab9a6396a0c4c2a74a923dbaca92c469fe38eb0bfde131b21321d75e2d155437f4f5b918c9becb327096dd5e72343f39c51565b6110f78be",
+        "error" : "s_a is the negated and therefore doesn't match R_a"
+    },
+    {
+        "adaptor_sig" : "024288a3e3d08ac29bab9a6396a0c4c2a74a923dbaca92c469fe38eb0bfde131b203127e25e61da3a9afbb0708eba169c1454f2a12785e3bd371b1a070d37c856b4a5adc9e4e988b4d30470fc94b838ebc202f6bf5fe66805cb957c89275cee34139d05ad315068f38bce87ffe269fed4cfb68438a536e64bda2d6bac695038b2499348a2a6fed030aa94952e36aaf002657e3ad5b1da9a2b90055675468e0011ee5",
+        "message_hash": "8131e6f4b45754f2c90bd06688ceeabc0c45055460729928b4eecf11026a9e2d",
+        "public_signing_key" : "03ef16dd7c75ca40cfeab2aa659f2201e857591df3de67494a4d1dae34587395e6",
+        "encryption_key" : "0254425b8f73754f16f758937fdab39a04ea61763f9c8b2ba8abe082882ddbf155",
+        "decryption_key" : "7f82dd63f4f75c33da444b72372be3aa43c0027a076bf9675eb7932695d127a4",
+        "signature": "bd775c1c2f753d6454659c695f3b3d57701c9f2be4b5dbd1c1997380d2550f8f1321d75e2d155437f4f5b918c9becb327096dd5e72343f39c51565b6110f78be",
+        "error": "signature R value does not match encrypted signature"
```

Comment:
Hmm I don't think this test vector makes sense. It makes sense to test this aspect of the code but this test doesn't do it since this test will just fail when you check of the decrypted signature is the same as the one in the JSON.

Reply:
There are now separate "recovery" tests for this.